src/Fleck/QueuedStream.cs

```diff
@@ -2,9 +2,127 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace Fleck
 {
+#if NETCORE
+    // TODO: Is this hack needed for .NET Core? If so, implement it.
+    public class QueuedStream : Stream
+    {
+        readonly Stream _stream;
+
+        public QueuedStream(Stream stream)
+        {
+            _stream = stream;
+        }
+
+        public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+        {
+            return _stream.CopyToAsync(destination, bufferSize, cancellationToken);
```

Comment:
awaiting would just wrap the CopyToAsync task with another task that does nothing but return the original result. Why do you think it should be wrapped?